pkg/utils/nodeclass/nodeclass.go

```diff
@@ -72,7 +72,7 @@ func NewSubnetSelectorTerms(subnetSelector map[string]string) (terms []v1beta1.S
 	for k, v := range subnetSelector {
 		switch k {
 		case "aws-ids", "aws::ids":
-			ids = strings.Split(strings.Trim(v, " "), ",")
+			ids = strings.Split(strings.ReplaceAll(v, " ", ""), ",")
```

Comment:
This trims spaces from the middle, which I don't think is what we want here. I'd prefer that we just split and then trim the whitespace off the ends of the result elements in the array

Reply:
Also, since this change with white spaces in the middle of values wasn't caught here, I'd like to see us having a functional test for that scenario. 

I doubt many users are putting spaces in the middle of their values or tags, but (if they are), this change would break them